### DISCUSSION TITLE
Guice missing implementaton for Materializer after upgrading to Play 3

### DISCUSSION THREAD

**Discussion by rafhuys-klarrio** [playframework/playframework#12918] [Q&A]
I add a few Http filters to my Play app. After upgrading from 2.9 to 3.0.5 I have this error

```
Exception encountered when invoking run on a nested suite - Unable to create injector, see the following errors:

1) [Guice/MissingImplementation]: No implementation for Materializer was bound.
```

I have a few filters:

`play.http.Filters += "Filters"`

```scala
class Filters @Inject()(defaultFilters: EnabledFilters, gzip: GzipFilter, log: LoggingFilter)
    extends DefaultHttpFilters(defaultFilters.filters :+ gzip :+ log: _*)

class LoggingFilter @Inject()(val metrics: Metrics, router: Router)(implicit val mat: Materializer, ec: ExecutionContext) extends Filter with LazyLogging {
```

So somewhere something is off. Does anyone have a idea of a working Play3 application with filters so I can check the diff of that project versus mine?

**Comment by mkurz** [comment]
I am pretty sure this is a problem in your setup. First, you config key should be lower case: `play.http.filters +=...`. 
Also, did you restart sbt after migrating the Play 3? Did you change all the configs and imports according to the migration guide?

**Comment by Squeng** [comment]
https://fixadat.com/ is a working Play 3 app
https://github.com/Squoss/Fixadat/tree/main/beapi
that uses various filters
https://github.com/Squoss/Fixadat/blob/main/beapi/conf/application.conf
including a custom one
https://github.com/Squoss/Fixadat/blob/main/beapi/app/filters/MessagesFilter.scala
